Source/WebCore/style/StyleScope.cpp

```diff
@@ -845,6 +845,15 @@ void Scope::didChangeViewportSize()
         return;
     resolver->clearCachedDeclarationsAffectedByViewportUnits();
 
+    if (customPropertyRegistry().invalidatePropertiesWithViewportUnits(m_document)) {
+        if (is<Document>(rootNode)) {
+            if (auto element = downcast<Document>(rootNode).documentElement())
+                element->invalidateStyleForSubtree();
+        } else if (is<Element>(rootNode))
+            downcast<Element>(rootNode).invalidateStyleForSubtree();
```

Comment:
`rootNode` here is always either a `ShadowRoot` or `Document`, neither of which is an `Element`. 

I don't think you need to do anything here in the shadow root case, invalidation the document by its scope is sufficient.